lib/watchdog/watchdog.go

```diff
@@ -0,0 +1,252 @@
+package watchdog
+
+import (
+	"context"
+	"errors"
+	"log"
+	"os"
+	"os/exec"
+	"os/signal"
+	"sync"
+	"sync/atomic"
+	"syscall"
+	"time"
+
+	"github.com/tarantool/tt/cli/process_utils"
+)
+
+// Watchdog manages a child process, ensuring reliable startup, automatic restarts on failure,
+// and graceful shutdown. It handles system signals, maintains PID file consistency,
+// and provides thread-safe operations for concurrent process management.
+type Watchdog struct {
+	// Command for launching and managing the child process.
+	cmd *exec.Cmd
+	// Timeout before restarting a failed process.
+	restartTimeout time.Duration
+	// Atomic flag for graceful shutdown.
+	shouldStop atomic.Bool
```

Comment:
This is a common code style in go. Please fix comments below.
```suggestion
	// cmd is a command for launching and managing the child process.
	cmd *exec.Cmd
	// restartTimeout is a timeout before restarting a failed process.
	restartTimeout time.Duration
	// shouldStop is an atomic flag for graceful shutdown.
	shouldStop atomic.Bool
```

Reply:
Done